apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/entity/blockchain/BlockImpl.java

```diff
@@ -380,69 +342,82 @@ public boolean checkSignature() {
     }
 
     @Override
-    public void setPrevious(Block block, HeightConfig config, Shard lastShard) {
+    public void setPrevious(Block[] threeLatestBlocksArray, HeightConfig config, Shard lastShard, int initialBlockHeight) {
```

Comment:
I guess that better to move such consensus logic to separate place

Reply:
Logic is moved into another component